sig-testing/flakiness-sla.md

```diff
@@ -0,0 +1,91 @@
+# Kubernetes Testing Flakiness SLA
+
+This document captures the expectations of the community about flakiness in
+our tests and our test infrastructure. It sets out an SLA (Service Level 
+Agreement) for flakiness in our tests, as well as actions that we will
+take when we are out of SLA.
+
+
+## Definition of a "Flake"
+
+We'll start by the definition of a _flake_. _Flakiness_ is defined for a
+complete run of one job in the pre-submit testing infrastructure 
+(e.g. pull-kubernetes-e2e-gke-gci) for a given pull request (PR).
+
+A pre-submit job's test result is considered to
+be a flake according to two criteria:
+
+1) it both fails and passes without any changes in the code
+or environment being tested.
+
+2) the PR in question doesn't cause the flake itself.
+
+### Measuring flakiness
+There are a number of challenges in monitoring flakiness. We expect that
+the metric will be heuristic in nature and we will iterate on it over time.
+Identifying all of the potential problems and ways to measure the metric
+are out of the scope of the document, but some currently known challenges
+are listed at the end of the document.
+
+## Flakiness SLA
+We will measure flakiness based on pull requests that are run through pre-submit
+PR testing. The metric that we will monitor is: Pre-submit flakes per PR. This metric
+will be calculated on a daily and weekly basis using:
+
+Sum(Flakes not caused by the PR) / Total(PRs)
```

Comment:
I believe that if we see the same test suite fail and then fix on multiple different PRs we have good confidence that it's flaky.  Likewise, if we see suites flake on docs-only PRs that's a good indication of flake. I don't believe it will be perfect, but I believe it will be close enough.

I've tried to capture this below in the doc.